### PR TITLE
Add CFG++ to DPM++ 2S Ancestral

### PR DESCRIPTION
**Pull request by yoinked-h**
I don't know if this is the correct formatting, but I decided to call it `dpmpp_2s_ancestral_pp` instead of the funnier `dpmpppp_2s_ancestral`

Example comparison v
image

(p.s. thank you to foxfu@tohoai for finding out how to make the conditioning work and not be swapped)

### CODE REVIEW

File: comfy/k_diffusion/sampling.py [hunk 1]
```diff
@@ -996,5 +996,45 @@ def post_cfg_function(args):
         dt = sigma_down - sigmas[i]
         x = x + d * dt
         if sigmas[i + 1] > 0:
+
```
Comment (comfyanonymous) [inline comfy/k_diffusion/sampling.py]: You accidentally removed code from this sampler.

Comment (yoinked-h) [reply comfy/k_diffusion/sampling.py]: ah, my bad; should be fixed now

**Comment by fgdfgfthgr-fox** [comment]
I wonder what's stopping this PR from merging. Tried it and it works fine for me.

**Comment by yoinked-h** [comment]
it probably is in the backlog, ive asked comfy in discord to check it so hopefully it gets merged now that its somewhat more up to date

**Pull request merged**